service/src/main/java/com/commercetools/pspadapter/payone/SessionHandler.java

```diff
@@ -0,0 +1,202 @@
+        } catch (final Exception e) {
+            return handleException(paymentId, retryCounter, e);
+        }
+        return handleException(paymentId, retryCounter,
+                new Exception("Unknown workflow error in PaymentHandler#handlePayment"));
+    }
+
+    private PayoneResult startSession(@Nonnull final String paymentId)
+            throws ConcurrentModificationException {
+
+        final PaymentWithCartLike paymentWithCartLike =
+                commercetoolsQueryExecutor.getPaymentWithCartLike(paymentId);
+
+        final String paymentInterface = paymentWithCartLike
+                .getPayment()
+                .getPaymentMethodInfo()
+                .getPaymentInterface();
+
+        if (!payoneInterfaceName.equals(paymentInterface)) {
+            final String errorMessage = format("Wrong payment interface name: expected '%s', found '%s' for the "
+                    + "commercetools Payment with id '%s'.", payoneInterfaceName, paymentInterface, paymentId);
+            return new PayoneResult(HttpStatusCode.BAD_REQUEST_400, errorMessage);
+        }
+        final PaymentMethodInfo paymentMethodInfo = paymentWithCartLike.getPayment().getPaymentMethodInfo();
+        if (!StringUtils.equals(paymentMethodInfo.getMethod(), INVOICE_KLARNA.getKey())) {
+            final String errorMessage = format("The session can only be started for the payment method '%s'",
+                    INVOICE_KLARNA.getKey());
+            return new PayoneResult(HttpStatusCode.BAD_REQUEST_400, errorMessage);
+        }
+        if (paymentWithCartLike.getPayment().getCustom() != null && StringUtils.isNotBlank(paymentWithCartLike.getPayment().getCustom().getFieldAsString(START_SESSION_RESPONSE))) {
+            final String errorMessage = format("The session was already started for the payment '%s'", paymentId);
+            return new PayoneResult(HttpStatusCode.BAD_REQUEST_400, errorMessage);
+        }
+        KlarnaRequestFactory requestFactory = new KlarnaRequestFactory(tenantConfig, new PayoneKlarnaCountryToLanguageMapper());
+        StartSessionRequestWithCart startSessionRequest = requestFactory.createStartSessionRequest(paymentWithCartLike);
+        final Map<String, Object> responseInfo = new HashMap<>();
+        final Map<String, String> response;
+        try {
+
+            response = payonePostService.executePost(startSessionRequest);
+        } catch (PayoneException paymentException) {
+            final String errorMessage = format("The 'startSession' Request to Payone failed for commercetools Payment" +
+                    " with id '%s'.", paymentWithCartLike.getPayment().getId());
+            logger.error(errorMessage, paymentException);
+            return new PayoneResult(HttpStatusCode.BAD_REQUEST_400, errorMessage);
+        }
+        String requestClientTokenName = "add_paydata[client_token]";
+        if (response.containsKey(requestClientTokenName)) {
```

Comment:
I wouldn't put this into the response. If this check fails, it shouldn't return 200.

Reply:
c5aaa4a60